Frameworks/Foundation/NSDecimal.mm

```diff
@@ -0,0 +1,640 @@
+                                     NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSCalculationError NSDecimalDivide(NSDecimal* result,
+                                   const NSDecimal* leftOperand,
+                                   const NSDecimal* rightOperand,
+                                   NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSString* NSDecimalString(const NSDecimal* dcm, id locale) STUB_METHOD {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSCalculationError NSDecimalMultiplyByPowerOf10(NSDecimal* result, const NSDecimal* number, short power, NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSCalculationError NSDecimalPower(NSDecimal* result, const NSDecimal* number, NSUInteger power, NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+void NSDecimalRound(NSDecimal* result, const NSDecimal* number, NSInteger scale, NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+}
+
+static void _InternalDecimalCopy(NSDecimal* destination, NSDecimalInternal* source) {
+    if (!source || !destination) {
+        return;
+    }
+    NSDecimalCopy(destination, &(source->decimal));
+    memcpy(destination->_mantissa, source->_mantissa, NSDecimalMaxSize * sizeof(unsigned short));
+}
+
+static NSComparisonResult _compareMantissa(NSDecimal* left, NSDecimal* right) {
+    // Note: All inputs have been normalized via NSDecimalNormalize
+    if (left->_length > right->_length) {
+        return NSOrderedDescending;
+    }
+
+    if (left->_length < right->_length) {
+        return NSOrderedAscending;
+    }
+
+    // same length
+    for (int index = right->_length - 1; index >= 0; index--) {
+        if (right->_mantissa[index] > left->_mantissa[index]) {
+            return NSOrderedAscending;
+        }
+
+        if (right->_mantissa[index] < left->_mantissa[index]) {
+            return NSOrderedDescending;
+        }
+    }
+
+    return NSOrderedSame;
+}
+
+static NSCalculationError _handleDifferenceOfSignAddition(NSDecimal* result,
+                                                          NSDecimal* left,
+                                                          NSDecimal* right,
+                                                          NSRoundingMode roundingMode) {
+    // Note: All inputs have been normalized via NSDecimalNormalize
+    // Since these are normalized and to account for unsigned subtraction we have to do mantissa based comparison.
+
+    NSComparisonResult res = _compareMantissa(left, right);
+
+    if (res == NSOrderedSame) {
+        SET_TO_ZERO(result);
+        return NSCalculationNoError;
+    }
+
+    if (res == NSOrderedAscending) {
```

Comment:
You could save some code duplication here if you do

NSDecimal* greater = right;
NSDecimal* lesser = left;

if (res == NSOrderedDescending) {
    std::swap(greater, lesser);
}

then just take the code from your first case and replace 'right' with 'greater' and 'left' with 'lesser'